spec/src/main/asciidoc/messaging-spec.adoc

```diff
@@ -26,3 +26,7675 @@ include::license-efsl.adoc[]
+significantly across products. There are major differences in the
+content and semantics of headers. Some products use a self describing,
+canonical encoding of message data; others treat data as completely
+opaque. Some products provide a repository for storing message
+descriptions that can be used to identify and interpret message content;
+others don’t.
+
+It would be quite difficult for Jakarta Messaging to capture the breadth
+ of this,sometimes conflicting, union of message models.
+
+=== Goals
+
+The Jakarta Messaging message model has the following goals:
+
+* Provide a single, unified message API
+* Provide an API suitable for creating messages that match the format
+used by existing, non-Jakarta Messaging applications
+* Support the development of heterogeneous applications that span
+operating systems, machine architectures, and computer languages
+* Support messages containing Java objects
+* Support messages containing Extensible Markup Language pages (see
+http://www.w3.org/XML).
+
+=== Jakarta Messaging messages
+
+Jakarta Messaging messages are composed of the following parts:
+
+* Header - All messages support the same set of header fields. Header
+fields contain values used by both clients and providers to identify and
+route messages.
+* Properties - In addition to the standard header fields, messages
+provide a built-in facility for adding optional header fields to a
+message.
+** Application-specific properties - In effect, this provides a
+mechanism for adding application specific header fields to a message.
+** Standard properties - Jakarta Messaging defines some standard 
+properties that are, in effect, optional header fields.
+** Provider-specific properties – Some Jakarta Messaging providers may 
+require the use of provider-specific properties. Jakarta Messaging 
+defines a naming convention for these.
+* Body - Jakarta Messaging defines several types of message body which 
+cover the majority of messaging styles currently in use.
+
+=== Message header fields
+
+The following subsections describe each Jakarta Messaging message header
+ field. Amessage’s complete header is transmitted to all Jakarta 
+Messaging clients that receive the message. Jakarta Messaging does not 
+define the header fields transmitted to non-Jakarta Messaging clients.
+
+==== JMSDestination
+
+The JMSDestination header field contains the destination to which the
+message is being sent.
+
+When a message is sent this value is ignored. After completion of the
+send it holds the Destination object specified by the sending method.
+
+When a message is received, its destination value must be equivalent to
+the value assigned when it was sent.
+
+==== JMSDeliveryMode
+
+The JMSDeliveryMode header field contains the delivery mode specified
+when the message was sent.
+
+When a message is sent this value is ignored. After completion of the
+send, it holds the delivery mode specified by the sending method.
+
+See Section  “” for more information.
```

Comment:
Missing reference?

Reply:
This reference was already missing in the original document.

Reply:
> Missing reference?

The original Messaging.adoc also does not have this reference ... But the file Messaging21.docx seems to have them =D

Would you like me to include them in this pull request?